docs/schematypes.md

```diff
@@ -585,13 +585,56 @@ const userSchema = new Schema({
 const User = mongoose.model('User', userSchema);
 ```
 
-To populate every `socialMediaHandles` entry's `oauth` property, you should populate
+To populate every `socialMediaHandles` entry's `oauth` property, you should populatex
 on `socialMediaHandles.$*.oauth`:
 
 ```javascript
 const user = await User.findOne().populate('socialMediaHandles.$*.oauth');
 ```
 
+<h4 id="uuid">UUID</h4>
+
+Mongoose also supports a UUID type that stores UUID v4 instances as [Node.js buffers](https://thecodebarbarian.com/an-overview-of-buffers-in-node-js.html).
+We recommend using [ObjectIds](#objectids) rather than UUIDs for unique document ids in Mongoose, but you may use UUIDs if you need to.
+
+In Node.js, a UUID is represented as an instance of `bson.Binary` type with a [getter](./tutorials/getters-setters.html) that converts the binary to a string when you access it.
+Mongoose stores UUIDs as [binary data with subtype 4 in MongoDB](https://www.mongodb.com/docs/manual/reference/bson-types/#binary-data).
+
+```javascript
+const schema = new mongoose.Schema({
+  docId: {
+    type: mongoose.Schema.Types.UUID // Can also do `type: 'UUID'`
+  }
+});
+const Test = mongoose.model('Test', schema);
+
+const doc = await Test.create({
+  docId: '09190f70-3d30-11e5-8814-0f4df9a59c41'
+});
+
+// 'string'
+console.log(typeof doc.docId);
+// true
+console.log(doc.toObject().docId instanceof mongoose.mongo.BSON.Binary);
+
+const rawDoc = await Test.collection.findOne({ _id: doc._id });
+// true
+console.log(rawDoc.toObject().docId instanceof mongoose.mongo.BSON.Binary);
+```
+
+To create UUIDs, we recommend using the [uuid npm package](https://www.npmjs.com/package/uuid).
+
+```javascript
+const { v4: uuidv4 } = require('uuid');
+
+const schema = new mongoose.Schema({
+  docId: {
+    type: 'UUID',
+    default: () => uuidv4()
+  }
+});
+```
```

Comment:
I don't think we should recommend using an external package now that we have a built-in way to achieve the same result. Starting from Node v14.17.0, we now have `const { randomUUID: uuidv4 } = require('crypto');`